### ISSUE TITLE
Turning off EPC paging

### ISSUE THREAD

**Issue by az-hasan** [intel/linux-sgx#411]
Hello,

I am running a program with various buffer sizes inside the enclave, and I can see performance slow down and then drop once the EPC limit is reached.
I understand that the EPC size is set to 128MB maximum. Once the contents of an enclave get larger than this, some pages will need to be swapped out. Is there a way to turn off this paging in SGX, to ensure the active pages are not swapped out?

Thank you

**Comment by ZongminGu** [comment]
If you turn off the swap feature, you could not successfully load the enclave(i.e. the sgx_loadenclave would fail), because you already know you need to swap some pages out in your case.
And there is no way to turn off the swap feature, if you do not change the driver code.

**Comment by az-hasan** [comment]
I am assigning the buffers inside the enclave using malloc, at runtime. If I stick to sizes small enough that I know will not trigger paging, will I be able to load the enclave, if I turn off swap?

Also, where in the driver code do I turn off paging? I am looking at sgx_page_cache.c, at a function called "sgx_swap_pages" - what must I change to turn off the swap feature?

**Comment by andyzyb** [comment]
If you only use small EPC (for example, 64MB), it likely will not trigger swapping.  if you disable swapping, you cannot load an Enclave is bigger than actual EPC size.  Do you plan to disable swapping, but also load a bigger enclave?

**Comment by az-hasan** [comment]
I see. So the pages of the EPC are managed by the Linux memory management system, and if I want to ensure that for the entire run of the program, the EPC pages are not swapped out, I need to turn off the Linux swap feature (with the risk of program crashing + not being able to load an Enclave bigger than the EPC).
I don't have to worry about changing anything in the sgx_driver, or change any SGX settings. Am I correct in my understanding?

**Comment by andyzyb** [comment]
Yes, I think it is correct. But turning swapping off may have risk for system stability.
